contracts/koh/contracts/KingAutomaton.sol

```diff
@@ -559,61 +690,16 @@ contract KingAutomaton {
     * @param r R portion of the signature
     * @param s S portion of the signature
     */
-  function claimSlot(
-    bytes32 pubKeyX,
-    bytes32 pubKeyY,
-    uint8 v,
-    bytes32 r,
-    bytes32 s
-  ) public {
-    uint256 slot = uint256(pubKeyX) % numSlots;
-    uint256 key = uint256(pubKeyX) ^ mask;
-
-    // Check if the key can take over the slot and become the new king.
-    require(key > minDifficulty && key > slots[slot].difficulty, "Low key difficulty");
-
-    // Make sure the signature is valid.
-    require(verifySignature(pubKeyX, pubKeyY, bytes32(uint256(msg.sender)), v, r, s), "Signature not valid");
-
+  function claimSlot(bytes32 pubKeyX, bytes32 pubKeyY, uint8 v, bytes32 r, bytes32 s) public {
```

Comment:
I would put the code here in an abstract base class instead of library as they are mostly pure view functionality (except for the "mint")